src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -383,6 +383,23 @@ static int lease_parse_domain(const uint8_t *option, size_t len, char **ret) {
         return 0;
 }
 
+static void filter_bogus_addresses(struct in_addr *l, size_t *n) {
+        size_t i, j;
+
+        /* Silently filter DNS/NTP servers supplied to us that do not make outside of the local scope. */
+
+        for (i = 0, j = 0; i < *n; i ++) {
+
+                if (in4_addr_is_null(l+i) ||
```

Comment:
Could this variable be called something else than `l`? `l+i` and `l[j]` are not the most legible. Calling it `addresses` would make the whole thing easier to read.